docs/en/engines/table-engines/mergetree-family/mergetree.md

```diff
@@ -439,6 +439,50 @@ Syntax: `ngrambf_v1(n, size_of_bloom_filter_in_bytes, number_of_hash_functions,
 - `number_of_hash_functions` — The number of hash functions used in the Bloom filter.
 - `random_seed` — The seed for Bloom filter hash functions.
 
+Users can create [UDF](/docs/en/sql-reference/statements/create/function.md) to estimate the parameters set of `ngrambf_v1`. Query statements are as follows:  
+
+```sql
+CREATE FUNCTION bfEstimateFunctions [ON CLUSTER cluster]   
+AS  
+(total_nubmer_of_all_grams, size_of_bloom_filter_in_bits) -> round((size_of_bloom_filter_in_bits / total_nubmer_of_all_grams) * log(2));   
+  
+CREATE FUNCTION bfEstimateBmSize [ON CLUSTER cluster]   
+AS  
+(total_nubmer_of_all_grams,  probability_of_false_positives) -> ceil((total_nubmer_of_all_grams * log(probability_of_false_positives)) / log(1 / pow(2, log(2))));  
+    
+CREATE FUNCTION bfEstimateFalsePositive [ON CLUSTER cluster]  
+AS   
+(total_nubmer_of_all_grams, number_of_hash_functions, size_of_bloom_filter_in_bytes) -> pow(1 - exp(-number_of_hash_functions/ (size_of_bloom_filter_in_bytes / total_nubmer_of_all_grams)), number_of_hash_functions);  
+  
+CREATE FUNCTION bfEstimateGramNumber [ON CLUSTER cluster]   
+AS  
+(number_of_hash_functions, probability_of_false_positives, size_of_bloom_filter_in_bytes) -> ceil(size_of_bloom_filter_in_bytes / (-number_of_hash_functions / log(1 - exp(log(probability_of_false_positives) / number_of_hash_functions))))
+
+```  
+To use those functions,we need to specify two parameter at least.
+For example, if there 4000 ngrams in the granule and we expect false positives to be less than 0.0001. The other parameters can be estimated by executing following queries:   
```

Comment:
Do you know how to estimate number of ngrams?

I came up with following query, but it's quite complex, so I don't think it's good to include.


```sql
-- concatenate all strings in each part, calculate ngrams, and then count the number of unique ones.

SELECT
    length( arrayDistinct( ngrams( arrayStringConcat( groupArray(s) ), 3 ) ) ) / p.marks AS ngrams_per_mark,
    _part
FROM t1
INNER JOIN system.parts AS p ON (t1._part = p.name) AND (p.table = 't1')
GROUP BY
    _part,
    p.marks

Query id: 20735b41-897a-422f-ac5a-6aec94873fbf

┌───ngrams_per_mark─┬─_part─────┐
│ 84.76923076923077 │ all_1_1_0 │
└───────────────────┴───────────┘
```

Reply:
The approach above can only estimate the number of ngrams for existing data. And the skipping indices only work for new data. This result can only be used to estimate the number. And it may mislead users if their String data is not fixed length in my opinion. So I think we should  let the user figure it out in their own way. How do you think ?
BTW, I have no idea to calculate the exact value with no fixed string.